src/pass/split_host_device.cc

```diff
@@ -192,8 +192,8 @@ class HostDeviceSplitter : public IRMutator {
         if (it != handle_data_type_.end()) {
           n->handle_data_type.Set(v, it->second);
         } else {
-          // int32 as a placeholder
-          n->handle_data_type.Set(v, make_const(UInt(32), 0));
+          // mark all the pointer argument 'restrict', use float32 as default type
```

Comment:
Can you give a case when we hit this codepath?

Reply:
remove this and update the codegen